test/cql-pytest/test_permissions.py

```diff
@@ -184,3 +184,74 @@ def test_udf_permissions_quoted_names(cassandra_bug, cql):
 
                 resources_with_execute = [row.resource for row in cql.execute(f"LIST EXECUTE OF {user}")]
                 assert f'<function {keyspace}.{weird_fun}(int)>' in resources_with_execute
+
+# Test that permissions set for user-defined functions are enforced
+# Tests for ALTER are separate, because they are qualified as cassandra_bug
+def test_grant_revoke_udf_permissions(cql):
+    schema = "a int primary key"
+    user = "cassandra"
+    with new_test_keyspace(cql, "WITH REPLICATION = { 'class': 'SimpleStrategy', 'replication_factor': 1 }") as keyspace:
+        with new_test_table(cql, keyspace, schema) as table:
+            fun_body_lua = "(i int) CALLED ON NULL INPUT RETURNS int LANGUAGE lua AS 'return 42;'"
+            fun_body_java = "(i int) CALLED ON NULL INPUT RETURNS int LANGUAGE java AS 'return 42;'"
+            fun_body = fun_body_lua
+            try:
+                with new_function(cql, keyspace, fun_body) as fun:
+                    pass
+            except:
+                fun_body = fun_body_java
+            with new_user(cql) as username:
+                with new_session(cql, username) as user_session:
+                    fun = "fun42"
```

Comment:
Can you please also write a test (probably a separate test - it will also be cassandra_bug), also with a weird name that needs quoting like?
You had a special test earlier about the quoting, but I think it's not enough. I shows the internal encoding, but not that this encoding can actually be used by our code.

Reply:
If I understand correctly, the `test_udf_permissions_quoted_names` was added for this